app/uk/gov/hmrc/gform/obligation/ObligationService.scala

```diff
@@ -17,30 +17,131 @@
 package uk.gov.hmrc.gform.obligation
 import java.text.SimpleDateFormat
 
+import cats.data.NonEmptyList
+import cats.kernel.Eq
+import cats.implicits._
+import uk.gov.hmrc.gform.auth.AuthService
+import uk.gov.hmrc.gform.auth.models.MaterialisedRetrievals
+import uk.gov.hmrc.gform.controllers.helpers.FormDataHelpers
 import uk.gov.hmrc.gform.gformbackend.GformConnector
 import uk.gov.hmrc.gform.sharedmodel._
+import uk.gov.hmrc.gform.sharedmodel.form.{ Form, FormId, UserData }
 import uk.gov.hmrc.gform.sharedmodel.formtemplate._
 import uk.gov.hmrc.http.HeaderCarrier
-import scala.concurrent.ExecutionContext
+
+import scala.concurrent.{ ExecutionContext, Future }
 
 class ObligationService(gformConnector: GformConnector) {
 
   val stringToDate = new SimpleDateFormat("yyyy-MM-dd")
 
-  def lookupObligationsMultiple(formTemplate: FormTemplate)(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
+  def lookupObligationsMultiple(
+    formTemplate: FormTemplate,
+    authService: AuthService,
+    retrievals: MaterialisedRetrievals,
+    form: Form)(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
+    val hmrcTaxPeriodIdentifiers = formTemplate.expandFormTemplateFull.allFCs.collect {
```

Comment:
Since `lookupObligationsMultiple` is called only from `lookupIfPossible` we can passed in `hmrcTaxPeriodIdentifiers : NonEmptyList[HmrcTaxPeriod]` as input parameter and do not calculate it second time (plus we already now that we need non empty list).